src/editor_transport.rs

```diff
@@ -65,27 +65,25 @@ pub fn start(config: &Config, initial_request: Option<String>) -> Result<EditorT
                     .spawn()
                 {
                     Ok(mut child) => {
-                        {
```

Comment:
> Maybe earlier Rust didn't allow to use `stdin` and `child.wait()` in the same scope since both use mutable references?

I think yes it didn't, moreover `child.wait()` is dropping `stdin`. At least [this](https://play.rust-lang.org/?version=stable&mode=debug&edition=2018&gist=d0eb3a32ece5ac6bdfad6036204faa81) playground didn't compile with rust 1.32. Here is [related](https://github.com/PacktPublishing/Mastering-RUST-Second-Edition/issues/1) question.

Reply:
I can add extra scope for backward compatibility.

Reply:
Thanks for clarifying, so it's really because of "non-lexical lifetimes". No need to add the scope IMO, because the behavior is not changed.